pkg/ui/handlers.go

```diff
@@ -62,11 +68,24 @@ func (a *API) uiFiles(w http.ResponseWriter, r *http.Request) {
 	a.fileServer.ServeHTTP(w, r)
 }
 
-func NewAPI(fileSystem fs.FS, logger logging.LoggerInterface) *API {
+func (a *API) getCSP(baseURL, kratosPublicURL string) string {
+	b, _ := url.Parse(baseURL)
+	k, _ := url.Parse(kratosPublicURL)
+	if k != nil && b != nil && k.Host != b.Host {
```

Comment:
Parsing the URLs may fail, this is swallowed here because the `BaseURL` is not a required parameter. Since this happens when the application starts, we could also panic to force the user to provide valid config.